packaging/npm/src/cli.ts

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env node
+
+import * as path from "path";
+import * as child_process from "child_process";
+
+const EXECUTABLE = "defang";
+
+function getPathToExecutable(): string {
+  let extension = "";
+  if (["win32", "cygwin"].includes(process.platform)) {
+    extension = ".exe";
+  }
+
+  const executablePath = path.join(__dirname, `${EXECUTABLE}${extension}`);
+  try {
+    return require.resolve(executablePath);
+  } catch (e) {
+    throw new Error(`Could not find application binary at ${executablePath}.`);
+  }
+}
+
+function run(): void {
+  const args = process.argv.slice(2);
+  const processResult = child_process.spawnSync(getPathToExecutable(), args, {
```

Comment:
js wrapper to use by npx or npm exec, this will call the **defang** binary with the arguments passed to the npx line. NPM installer should do the linking of the defang package to the user PATH to the cli.js to execute.
e.g. npx defang login